cache/interface.go

```diff
@@ -2,25 +2,37 @@ package cache
 
 import (
 	"github.com/filecoin-project/go-indexer-core"
-	peer "github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/peer"
 	"github.com/multiformats/go-multihash"
 )
 
 // Interface is the interface any result cache used in the indexer. A result
 // cache only keeps results for completed requests, and may only store a
 // limited quantity of values.
 type Interface interface {
-	// Get retrieves a slice of indexer.Value for a multihash
-	Get(multihash.Multihash) ([]indexer.Value, bool, error)
-	// Put stores an additional indexer.Value for a multihash if the value is not already stored
-	Put(multihash.Multihash, indexer.Value) (bool, error)
-	// PutMany stores an indexer.Value for multiple multihashes
-	PutMany([]multihash.Multihash, indexer.Value) error
-	// Remove removes an indexer.Value for a multihash
-	Remove(multihash.Multihash, indexer.Value) (bool, error)
-	// RemoveMany removes an indexer.Value from multiple multihashess
-	RemoveMany([]multihash.Multihash, indexer.Value) error
-	// RemoveProvider removes all entries for specified provider.  This is used
-	// when a provider is no longer indexed by the indexer.
-	RemoveProvider(peer.ID) error
+	// Get retrieves a slice of Values that a multihash index maps to.
+	Get(multihash.Multihash) ([]indexer.Value, bool)
+
+	// Put stores a Value and adds a mapping from each of the given multihashs
+	// to that Value.  If the Value has the same ProviderID and ContextID as a
+	// previously stored Value, then update the metadata in the stored Value
+	// with the metadata from the provided Value.  Call Put without any
+	// multihashes to only update existing values.
+	Put(indexer.Value, ...multihash.Multihash) int
+
+	// Remove removes the mapping of each multihash to the specified Value.
+	// Returns the number of indexe mappings removed.
+	Remove(indexer.Value, ...multihash.Multihash) int
+
+	// RemoveProvider removes all values for the specified provider, and all
+	// indexes that map to those values.  This is used when a provider is no
+	// longer indexed by the indexer.  Returns the number of indexes that were
```

Comment:
I also learned in school this style (two spaces after period) for text entry 😄

Reply:
Because emacs recognizes that as a next sentence, so I can move forward and back to the next sentence easily.  Also, 
 somewhat of a habit leftover from grade school when children were taught to put two spaces after a period, whether writing or typing.  Mostly the first reason though.  I can ditch it if it is not the way to do things.